src/audio/copier/copier.c

```diff
@@ -715,6 +715,9 @@ static void copier_free(struct comp_dev *dev)
 		rfree(cd->hd);
 	}
 
+	if (cd->multi_endpoint_buffer)
+		buffer_free(cd->multi_endpoint_buffer);
```

Comment:
btw, I just checked that `buffer_free(NULL)` is ok, which is good, because it makes it consistent with `free(3)`, `rfree()`, etc. So in principle the above `if` isn't needed. But no, don't think it's compulsory to remove it